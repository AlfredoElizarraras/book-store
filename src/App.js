import './App.css';
import React from 'react';

const App = () => (
  <div>
    Book store app.
  </div>
);

export default App;
