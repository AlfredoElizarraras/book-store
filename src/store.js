import { createStore } from 'redux';
import { devToolsEnhancer } from 'redux-devtools-extension';
import rootReducer from './reducers';
import initBooks from './utils/data';
import { loadState } from './utils/localStorage';

const localState = loadState();
const books = localState && localState.books ? localState.books : initBooks.books;

/* eslint no-underscore-dangle: 0 */
export default createStore(
  rootReducer,
  { books },
  devToolsEnhancer(),
);
