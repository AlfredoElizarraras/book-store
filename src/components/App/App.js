import './App.css';
import React from 'react';
import BookList from '../../containers/BookList';
import BookForm from '../BookForm';
import Navbar from '../Navbar';

const App = () => (
  <div>
    <Navbar />
    <BookList />
    <BookForm />
  </div>
);

export default App;
