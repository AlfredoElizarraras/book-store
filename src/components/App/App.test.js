import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from './App';
import store from '../../store';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Provider store={store}><App /></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('Has a valid snapshot', () => {
    const component = renderer.create(<Provider store={store}><App /></Provider>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
