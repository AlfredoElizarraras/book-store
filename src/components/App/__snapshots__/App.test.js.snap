// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Has a valid snapshot 1`] = `
<div>
  <form>
    <label
      htmlFor="title"
    >
      Book Title:
      <input
        id="title"
        name="title"
        onChange={[Function]}
        type="title"
        value=""
      />
    </label>
    <label
      htmlFor="categories"
    >
      Choose a car:
      <select
        id="categories"
        name="categories"
        onChange={[Function]}
        value="Action"
      >
        <option
          value="Action"
        >
          Action
        </option>
        <option
          value="Biography"
        >
          Biography
        </option>
        <option
          value="History"
        >
          History
        </option>
        <option
          value="Horror"
        >
          Horror
        </option>
        <option
          value="Kids"
        >
          Kids
        </option>
        <option
          value="Learning"
        >
          Learning
        </option>
        <option
          value="Sci-Fi"
        >
          Sci-Fi
        </option>
      </select>
      <button
        onClick={[Function]}
        type="button"
      >
        Add Book
      </button>
    </label>
  </form>
  <table>
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Title
        </th>
        <th>
          Category
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          58109aa4-c849-4280-9d8a-8720ac67376a
        </td>
        <td>
          Book 1
        </td>
        <td>
          Action
        </td>
      </tr>
      <tr>
        <td>
          da1340a4-d28d-470a-8b90-b6e81e92f79a
        </td>
        <td>
          Book2
        </td>
        <td>
          Learning
        </td>
      </tr>
      <tr>
        <td>
          a7726433-a81e-4392-8ef2-bf2cb5fd7f1a
        </td>
        <td>
          Book3
        </td>
        <td>
          Horror
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
